src/openrct2/scripting/bindings/object/ScInstalledObject.hpp

```diff
@@ -0,0 +1,196 @@
+            dukglue_register_property(ctx, &ScInstalledObject::type_get, nullptr, "type");
+            dukglue_register_property(ctx, &ScInstalledObject::sourceGames_get, nullptr, "sourceGames");
+            dukglue_register_property(ctx, &ScInstalledObject::legacyIdentifier_get, nullptr, "legacyIdentifier");
+            dukglue_register_property(ctx, &ScInstalledObject::authors_get, nullptr, "authors");
+            dukglue_register_property(ctx, &ScInstalledObject::name_get, nullptr, "name");
+        }
+
+    private:
+        std::string path_get() const
+        {
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                return installedObject->Path;
+            }
+            return {};
+        }
+
+        std::string generation_get() const
+        {
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                if (installedObject->Generation == ObjectGeneration::DAT)
+                    return "dat";
+                else
+                    return "json";
+            }
+            return {};
+        }
+
+        std::vector<std::string> sourceGames_get() const
+        {
+            std::vector<std::string> result;
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                for (const auto& sourceGame : installedObject->Sources)
+                {
+                    result.push_back(std::string(ObjectSourceGameToString(sourceGame)));
+                }
+            }
+            return result;
+        }
+
+        std::string type_get() const
+        {
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                return std::string(ObjectTypeToString(EnumValue(installedObject->Type)));
+            }
+            return {};
+        }
+
+        std::string identifier_get() const
+        {
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                if (installedObject->Generation == ObjectGeneration::DAT)
+                {
+                    return ObjectEntryDescriptor(installedObject->ObjectEntry).ToString();
+                }
+                else
+                {
+                    return installedObject->Identifier;
+                }
+            }
+            return {};
+        }
+
+        std::string legacyIdentifier_get() const
+        {
+            auto installedObject = GetInstalledObject();
+            if (installedObject != nullptr)
+            {
+                if (!installedObject->ObjectEntry.IsEmpty())
+                {
+                    return std::string(installedObject->ObjectEntry.GetName());
+                }
+            }
+            return {};
```

Comment:
I assume `{}` turns into `null` on the JS side?

Reply:
No this would be empty string.

Reply:
Oh in that case I'd say `null` would be preferable for objects without a DAT identifier.